### PR TITLE
Adding documentation on PSATD-JRhom

### PR DESCRIPTION
**Pull request by oshapoval**
Adding documentation on PSATD-JRhom.

### CODE REVIEW

File: Docs/source/theory/pic.rst [hunk 1]
```diff
@@ -434,6 +434,85 @@ of this model can be found in the section
...
+   Y_3 & = \frac{S(8- c^2k^2\delta t^2 ) - 4ck\delta t(1+C)}{2c^2 k^2 \delta t^2},
+   \\
+   Y_4 &= (1-C), \ Y_5 = \frac{(1+C) ck\delta t - 2S}{2ck \delta t}.
+   \end{aligned}
+
+Here, :math:`\mathbf{a_J}, \mathbf{b_J}, \mathbf{c_J}, a_{\rho}, b_{\rho}, c_{\rho}` are polynomial coefficients based on the time dependencies of the current  and charge densities, given in Table I:
+
+.. _fig-j_rho_table:
+
+.. figure:: j_rho_table.png
```
Comment (ax3l) [inline Docs/source/theory/pic.rst]: Can we store this on gist.github.com and add the raw link here?

Comment (EZoni) [reply Docs/source/theory/pic.rst]: This question ("how do we add figures to the docs?") comes up often.
Shall we add a few lines of guidance in our docs, somewhere here where we talk about our docs?
https://github.com/ECP-WarpX/WarpX/blob/development/Docs/source/developers/documentation.rst

**Comment by EZoni** [comment]
Discussed offline with @oshapoval: we keep this in the field solver section for now, since we are going to refactor the whole theory section anyways in #5258.

File: Docs/source/theory/pic.rst [hunk 2]
```diff
@@ -434,6 +434,85 @@ of this model can be found in the section
 
 .. _current_deposition:
 
+Pseudo Spectral Analytical Time Domain with arbitrary charge and current-density time dependencies (PSATD-JRhom)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In :cite:`pt-shapovalPRE2024` we introduce a formulation of the particle-in-cell (PIC) method for the modeling of relativistic plasmas, which leverages the ability of the pseudo-spectral analytical time-domain solver (PSATD) to handle arbitrary time dependencies of the charge and current densities during one PIC cycle (applied to second order polynomial dependencies here).
```
Comment (EZoni) [inline Docs/source/theory/pic.rst]: I think the formulas below, with the `a`, `b` and `c` coefficients, are valid for all time dependencies *up* to quadratic, and not quadratic only, right?
```suggestion
In :cite:`pt-shapovalPRE2024` we introduce a formulation of the particle-in-cell (PIC) method for the modeling of relativistic plasmas, which leverages the ability of the pseudo-spectral analytical time-domain solver (PSATD) to handle arbitrary time dependencies of the charge and current densities during one PIC cycle (up to second-order polynomial dependencies here).
```

File: Docs/source/theory/pic.rst (hunk 1 above)
Comment (EZoni) [reply Docs/source/theory/pic.rst]: @oshapoval
Would you document the few steps you followed to add the image with gist.github.com and add it somewhere in the docs page linked in the message above? I think we can easily include it in this PR.

**Review by EZoni — APPROVED** [review]

**Pull request merged**